document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresCollection.java

```diff
@@ -134,7 +134,36 @@ public CreateResult create(Key key, Document document) throws IOException {
 
   @Override
   public BulkUpdateResult bulkUpdate(List<BulkUpdateRequest> bulkUpdateRequests) throws Exception {
-    throw new UnsupportedOperationException();
+
+    List<BulkUpdateRequest> requestsWithFilter =
+        bulkUpdateRequests.stream()
+            .filter(req -> req.getFilter() != null)
+            .collect(Collectors.toList());
+
+    List<BulkUpdateRequest> requestsWithoutFilter =
+        bulkUpdateRequests.stream()
+            .filter(req -> req.getFilter() == null)
+            .collect(Collectors.toList());
+
+    try {
+      long totalUpdateCountA = bulkUpdateRequestsWithFilter(requestsWithFilter);
```

Comment:
I thought we were going with the late exception as discussed yesterday. We can take care of it as follow-up PR.